pkg/resources/reconciler_test.go

```diff
@@ -576,6 +637,36 @@ func TestReconciler_ReconcileNamespace(t *testing.T) {
 			if tc.ExpectedStatus != phase {
 				t.Fatal("expected phase ", tc.ExpectedStatus, " but got ", phase)
 			}
+			labelExists, labelIsTrue, err := verifyNSLabelExistsAndIsTrue(tc.client)
+			if err != nil {
+				t.Fatal("error when verifying namespace label exists")
+			}
+			if !labelExists && tc.ExpectLabel {
```

Comment:
if we have a tc.ExpectLabel bool and some of the tests cases use it as false, we should also have a condition here to check it, something like if labelExists and !tc.ExpectedLabel then fail

Reply:
other than that, these unit tests look good to me :) :+1: